frontend/src/metabase/visualizations/components/ScalarValue/utils.unit.spec.ts

```diff
@@ -61,11 +63,61 @@ describe("findSize", () => {
   });
 
   it("returns the min size if text cannot fit into the target width", () => {
-    measureTextSpy.mockImplementation(createMockMeasureText(120));
+    measureTextSpy.mockImplementation(createMockMeasureText(120, 50));
 
     const size = findSize({
       ...defaults,
       targetWidth: 100,
+      targetHeight: 100,
+      step: 0.2,
+      min: 2,
+      max: 5,
+    });
+
+    expect(size).toEqual("2rem");
+  });
+
+  it("returns the max size if when text height is smaller than the target height", () => {
+    measureTextSpy.mockImplementation(createMockMeasureText(50, 100));
+
+    const size = findSize({
+      ...defaults,
+      targetWidth: 100,
+      targetHeight: 100,
+      step: 0.2,
+      min: 2,
+      max: 5,
+    });
+
+    expect(size).toEqual("5rem");
+  });
+
+  it("returns the first size with which text height is smaller than the target height", () => {
+    measureTextSpy
+      .mockImplementationOnce(createMockMeasureText(50, 120))
+      .mockImplementationOnce(createMockMeasureText(50, 110))
+      .mockImplementationOnce(createMockMeasureText(50, 100)) // this is the one we want
+      .mockImplementationOnce(createMockMeasureText(0, 90));
```

Comment:
Similiar to my previous comment, I think we should kind of proportionally, mock the width and height. They don't need to maintain the exact aspect ratio, but at least they need to shrink together.

Also, I'm not sure if the last line that sets the width to `0` is a typo or not.

Reply:
> Also, I'm not sure if the last line that sets the width to 0 is a typo or not.

Yes, good catch :+1: 
f0219dd4d85a2c4e01ac6d71bcf2ed1dd0db2505

Reply:
c88f7c77f9b474a5c28325b3a74d314f56cc6b10